bundles/org.openhab.binding.ism8/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="ism8"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="device">
```

Comment:
The current definition doesn't give the user the option to create the channels in the ui, but the user must use text configuration. By adding `extensible` and defining `channel-type` like `switch` and `number` (I used arbitrary names) and specify in xml how such channel looks the user can create the channels via the ui.
```suggestion
	<thing-type id="device" extensible="switch, number">
```

Reply:
For this feature I need a bit more support. I was adapting the code above, but adding a channel in the UI isn't possible. I'm getting the following situation:
![grafik](https://user-images.githubusercontent.com/57272516/93028592-1d078680-f615-11ea-8c55-39ae51038ff5.png)

Is there any reference to check how this principle works? 
So far I was mainly focusing on adding the required information into the `ism8.things` file. Please check the example in the README.md. The binding specific configuration settings are: id, type, write (optionally).

Reply:
You need to add a channel-type definition in the xml. The values in `extensible` refer to this. For example:

```

<thing-type id="device" extensible="switch-readonly">

.....

<channel-type id="switch-readonly">
		<item-type>Switch</item-type>
		<label>Readonly Switch</label>
		<description> text here </description>
                <state readOnly="true" pattern="%s"/>
		<config-description>
			<parameter name="id" type="int" required="true">
				<label>Id</label>
                                <description> text here </description>
                       </parameter>
			<parameter name="type" type="text" required="true">
				<label>Type</label>
                                <description> text here </description>
                       </parameter>
		</config-description>
</channel-type>
```

While writing this I noticed you have a parameter to indicate the channel can write to. But this should really be modeled by using 2 different channel types. Where you set the state to readonly false or true. That way the user interface can disable it from input.

The user In the things file should refer to these channel-types instead of directly creating it using the channel type instead of the item type:
```
 Thing ism8:device:heater "Wolf Heizung"         [portNumber=12004]
    {
        Type switch-readonly : DpId001 "Störung Heizgerät"            [id=1, type="1.001"]
```
```